src/CADL.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -370,9 +381,15 @@ export function createModelForService(
         }
     }
 
-    const usages = getUsages(program, convenienceOperations);
-    setUsage(usages, modelMap);
-    setUsage(usages, enumMap);
+    if (generateAllModels) {
```

Comment:
I think we might have a conflict here with what `getAllSchemas` is trying to do in tcgc (typespec-client-generation-core).  Can we make sure we aren't introducing duplicate logic and if the logic in `getAllSchemas` isn't sufficient lets investigate how to make it so other languages can also take advantage of this.